plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -491,17 +530,26 @@ private static String getGeneratedColumnExpressions(JsonNode node)
         return generationExpression == null ? null : generationExpression.asText();
     }
 
-    public static Map<String, String> getCheckConstraints(MetadataEntry metadataEntry)
+    public static Map<String, String> getCheckConstraints(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
     {
+        if (protocolEntry.supportsWriterFeature() && !protocolEntry.writerFeaturesContains("checkConstraints")) {
+            return ImmutableMap.of();
+        }
         return metadataEntry.getConfiguration().entrySet().stream()
                 .filter(entry -> entry.getKey().startsWith("delta.constraints."))
                 .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
-    public static boolean changeDataFeedEnabled(MetadataEntry metadataEntry)
+    public static Optional<Boolean> changeDataFeedEnabled(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
```

Comment:
Why is it 3-state logic now?
Isn't that CDF is either enabled or not? What does " 🤷‍♂️ " mean here?

Reply:
Actually, even the previous logic is 3-state because MetadataEntry class had `Optional<Boolean> isChangeDataFeedEnabled()` method. I guess the motivation of the 3-state is hiding `change_data_feed_enabled` table properties in `SHOW CREATE TABLE` when it's undefined. 

cc: @homar